File: app/src/main/java/soup/movie/ui/detail/DetailContract.java
package soup.movie.ui.detail;

import android.support.annotation.NonNull;

import soup.movie.data.soup.model.Movie;
import soup.movie.ui.BaseContract;

class DetailContract {

    interface Presenter extends BaseContract.Presenter<View> {
        void requestData(@NonNull Movie movie);
        void requestData(@NonNull String code, @NonNull Movie movie);
    }

    interface View extends BaseContract.View {
        void render(@NonNull DetailUiModel uiModel);
    }
}

File: app/src/main/java/soup/movie/ui/detail/DetailContract.kt
package soup.movie.ui.detail

import android.net.Uri
import soup.movie.ui.LegacyBaseContract

interface DetailContract {

    interface Presenter : LegacyBaseContract.Presenter<View> {

        fun requestShareImage(url: String)
    }

    interface View : LegacyBaseContract.View {

        fun render(viewState: DetailViewState)

        fun doShareImage(imageUri: Uri, mimeType: String)
    }
}
